.pre-commit-config.yaml

```diff
@@ -0,0 +1,31 @@
+repos:
+    -   repo: https://github.com/pycqa/isort
+        rev: 5.8.0
```

Comment:
Is this a `>=` or `==`?

Reply:
`==`

Reply:
That's not very useful since each version of flake8 adds new checks.

Reply:
True. I just specified 3.8 since it's the minimum supported version (for `flake8`) according to `requirements.txt`.

Reply:
I don't have strong opinions about us using it. For me `pre-commit` was a convenient tool to catch style and formatting errors on my side rather than push it to the CI workflow and find out about it later. My original goal was to provide this option should any current/future contributors feel the same way and **if** it's harmless to maintain, add it to the repo.

As I mentioned earlier the one con to `pre-commit` is that any major changes to `pyproject.toml` will need to be reflected in the `pre-commit` config. 

Updated black version? Need to do that here. Added a new dependency that supports type hints? Need to add that here.
To be fair, this can be caught pretty easily but yeah this is something to be considered.

Truth be told, if the final call is to close this PR and refer users to it when new contributors keep hitting style/formatting errors that honestly fine too.